052-record-reconciled-versions.md

```diff
@@ -0,0 +1,190 @@
+# Record Reconciled Version in Kafka Custom Resource status
+
+For the `Kafka` Custom Resource it is often unclear when an upgrade has truly finished, whether all the brokers have rolled and to which version of the operator a cluster has been reconciled with and which version of Kafka.
+The following proposal puts forward a mechanism to make it clear in the `Kafka` custom resources status whether an upgrade is complete, making it easy to programmatically check for upgrade completion. This could potentially be used/adopted in the upgrade system tests or upgrade verification steps for a user, simplifying the code and making for a better upgrade UX story.
+
+## Current situation
+
+Currently the only way to know if an upgrade has finished to completion is to check is all the zookeeper and kafka brokers have rolled.
+https://github.com/strimzi/strimzi-kafka-operator/blob/2a1fdf9d8695bb22a2bf977b0ba5414291530207/systemtest/src/test/java/io/strimzi/systemtest/upgrade/KafkaUpgradeDowngradeST.java#L292
+Additionally, if a user updates the Kafka version, a user needs to check if their Kafka version has been updated by inspecting the Kafka image/Jar version that the Brokers are running with, and this could require two rolls of the brokers, not just a single roll.
+Though this works, from a user standpoint it is far from ideal as it assumes a level of user knowledge about the cluster that may not be appropriate for all users.
+
+## Motivation
+
+This proposal allows a user to rely on the `Kafka` custom resources as their single source of truth, instead of watching/querying the individual kafka/zookeeper pods.
+This proposal lays out all the information that is required, and at what points in the reconcile this information will be used. This information is already readily available within the Reconcile loop, so orchestrating and figuring out if a component has been fully upgraded will be a case of propagating this information into the Custom Resource via status fields.
+
+## Proposal
+
+This proposal will cover the `Kafka` custom resource but could apply to other CR types that use `StrimziPodSets` such as `KafkaConnect` and `KafkaMirrorMaker2`, but could be extended to include other Custom Resources if there would be value in it.
```

Comment:
```suggestion
This proposal will cover the `Kafka` custom resource, but could apply to other CR types that use `StrimziPodSets` such as `KafkaConnect` and `KafkaMirrorMaker2`. It could be extended to include other custom resources if there is value in it.
```